bpf/process/bpf_process_event.h

```diff
@@ -612,18 +603,12 @@ __event_get_task_info(struct msg_execve_event *msg, __u8 op, bool walker,
 		if (subsys) {
 			probe_read(&cgrp, sizeof(cgrp), _(&subsys->cgroup));
 			if (cgrp) {
-				probe_read(&kn, sizeof(cgrp->kn), _(&cgrp->kn));
-				if (kn) {
-					probe_read(&name, sizeof(name),
-						   _(&kn->name));
-					if (name) {
-						probe_read_str(
-							msg->kube.docker_id,
-							DOCKER_ID_LENGTH, name);
-					} else {
-						curr->flags |=
-							EVENT_DOCKER_NAME_ERR;
-					}
+				if (BPF_CORE_READ_INTO(&name, cgrp, kn, name) ==
+				    0) {
+					probe_read_str(msg->kube.docker_id,
+						       DOCKER_ID_LENGTH, name);
+				} else {
+					curr->flags |= EVENT_DOCKER_NAME_ERR;
```

Comment:
Thanks!

Would it make sense to use BPF_CORE_READ_STR_INTO for the final string?

Also, I wonder if it would make sense to also wrap the cgroups / subsys reads above into such a call. This would mean that we wouldn't not be able to distinguish the different errors (e.g., EVENT_DOCKER_SUBSYS_ERR) but I think that's fine.

Reply:
yep, it's possible to wrap all of them. I found tetragon records those errors, so I didn't change them. 

I'll review related error handlers in golang side to check if it's safe to remove these error flags.

Reply:
@kkourt In the golang part, these errors are defined at

https://github.com/cilium/tetragon/blob/b87b5d65ac1e92fc43cd6c3d85f6c37e4c1ae490/pkg/api/flags.go#L77

and decoded at 

https://github.com/cilium/tetragon/blob/b87b5d65ac1e92fc43cd6c3d85f6c37e4c1ae490/pkg/reader/exec/exec.go#L57

emmm, don't know if it is a good idea to remove them...

Reply:
Thanks for looking into this!

I'd be personally OK with keeping a single error for all different ways that getting the docker id would fail. @jrfastab what do you think?

Reply:
We can leave that as a followup, I guess.